core/src/tracker.ts

```diff
@@ -1,69 +1,91 @@
 import type { Notifier } from './types';
-import getMaker from './maker';
+import { ethers } from 'ethers';
 import { v4 as uuidv4 } from 'uuid';
+import truncateText from './helpers/truncateText';
 
-const trackTransaction = async function (transactionPromise: Promise<any>, notifier?: Notifier): Promise<string> {
-    const maker = await getMaker();
+const DEFAULT_NOTIFICATION_DURATION = 3;
+const NUMBER_OF_BLOCKS_TO_CONFIRM = 5;
+
+const trackTransaction = async function (
+    transactionPromise: Promise<ethers.ContractTransaction>,
+    notifier?: Notifier,
+    confirmTransaction = true
+): Promise<string> {
     const messageId = uuidv4();
-    const transactionManager = maker.service('transactionManager');
+
+    // Display initial message
     if (notifier) {
```

Comment:
Instead of checking the `notifier` at every point you use it, you could also set it up to always work in the beginning of the function. Looking at the types, something like:
```ts
const notifier = notifier || (messageType, messageContent: MessageContent) => {}
```

should allow you to use the notifier care free throughout this function (: